src/battle_script_commands.c

```diff
@@ -11264,11 +11265,152 @@ static void Cmd_various(void)
+            battlerId = gBattlerTarget = BATTLE_PARTNER(gActiveBattler);
+
+        // Get amount to heal.
+        switch (healParam)
+        {
+            case ITEM6_HEAL_HP_FULL:
+                healAmount = maxHP;
+                break;
+            case ITEM6_HEAL_HP_HALF:
+                healAmount = maxHP / 2;
+                break;
+            case ITEM6_HEAL_HP_QUARTER:
+                healAmount = maxHP / 4;
+                break;
+            default:
+                healAmount = healParam;
+                break;
+        }
+        if (hp + healAmount > maxHP)
+            healAmount = maxHP - hp;
+        
+        // Heal is applied as move damage if battler is active.
+        if (battlerId != MAX_BATTLERS_COUNT && hp != 0)
+        {
+            gBattleMoveDamage = -healAmount;
+        }
+        else
+        {
+            hp += healAmount;
+            SetMonData(&party[gBattleStruct->itemPartyIndex[gActiveBattler]], MON_DATA_HP, &hp);
+
+            // Revived battlers on the field need to be brought back.
+            if (gBattleTypeFlags & BATTLE_TYPE_DOUBLE && battlerId != MAX_BATTLERS_COUNT)
+            {
+                gAbsentBattlerFlags &= ~gBitTable[battlerId];
+                gBattleCommunication[MULTIUSE_STATE] = TRUE;
+            }
+        }
+        PREPARE_SPECIES_BUFFER(gBattleTextBuff1, GetMonData(&party[gBattleStruct->itemPartyIndex[gActiveBattler]], MON_DATA_SPECIES));
+        break;
+    }
+    case VARIOUS_ITEM_CURE_STATUS:
+    {
+        VARIOUS_ARGS();
+        struct Pokemon *party = (side == B_SIDE_PLAYER) ? gPlayerParty : gEnemyParty;
+        
+        // Heal Status1 conditions.
+        HealStatusConditions(&party[gBattleStruct->itemPartyIndex[gActiveBattler]], gBattleStruct->itemPartyIndex[gActiveBattler], GetItemStatus1Mask(gLastUsedItem), gActiveBattler);
+        
+        // Heal Status2 conditions if battler is active.
+        if (gBattleStruct->itemPartyIndex[gActiveBattler] == gBattlerPartyIndexes[gActiveBattler])
+            gBattleMons[gBattleStruct->itemPartyIndex[gActiveBattler]].status2 &= ~GetItemStatus2Mask(gLastUsedItem); 
+        else if (gBattleTypeFlags & BATTLE_TYPE_DOUBLE
+                 && gBattleStruct->itemPartyIndex[gActiveBattler] == gBattlerPartyIndexes[BATTLE_PARTNER(gActiveBattler)])
+        {
+            gBattleMons[gActiveBattler].status2 &= ~GetItemStatus2Mask(gLastUsedItem); 
+            gBattlerTarget = BATTLE_PARTNER(gActiveBattler);
+            if (GetItemStatus1Mask(gLastUsedItem) & STATUS1_SLEEP)
+                gBattleMons[gActiveBattler].status2 &= ~STATUS2_NIGHTMARE;
+        }
+        
+        PREPARE_SPECIES_BUFFER(gBattleTextBuff1, GetMonData(&party[gBattleStruct->itemPartyIndex[gActiveBattler]], MON_DATA_SPECIES));
+        break;
+    }
+    case VARIOUS_ITEM_INCREASE_STAT:
+    {
+        VARIOUS_ARGS();
+        u16 statId = GetItemEffect(gLastUsedItem)[1];
+        u16 stages = (B_X_ITEMS_BUFF >= GEN_7) ? 2 : 1;
```

Comment:
for the WL PR I get the appropriate X Item stage by `ItemId_GetHoldEffectParam`. If we agree that it is fine to misuse that var for this, then I would suggest adding `X_ITEM_STAGE` to the `holdEffectParam` of the xItems and move the definition of `X_ITEM_STAG`E in pokemon.c to an appropriate header file.

Reply:
Sure, will do!